lib/Sema/CSSimplify.cpp

```diff
@@ -4497,8 +4497,21 @@ bool ConstraintSystem::repairFailures(
       break;
     }
 
+    auto *parentLoc = getConstraintLocator(anchor, path);
+
     if ((lhs->is<InOutType>() && !rhs->is<InOutType>()) ||
         (!lhs->is<InOutType>() && rhs->is<InOutType>())) {
+      // Since `FunctionArgument` as a last locator element represents
+      // a single parameter of the function type involved in a conversion
+      // to another function type, see `matchFunctionTypes`. If there is already
+      // a fix for the this convertion, we can just ignore individual function
+      // argument in-out mismatch failure by considered this fixed increasing
+      // the score.
+      if (hasFixFor(parentLoc)) {
+        increaseScore(SK_Fix);
```

Comment:
`repairFailures` shouldn't increase a score like this because it's logically attached to the fix when it's applied in `simplifyFixConstraint`. If increasing score in necessary I'd suggest we let AllowInOutConversion be added to the fixes and in `simplifyFixConstraint` re-check the condition and increase impact of the fix.

Reply:
We just simply need to considered this fixed, no need to increase the score :)

Reply:
Great!